exec/java-exec/src/main/java/org/apache/drill/exec/planner/physical/visitor/LateralUnnestRowIDVisitor.java

```diff
@@ -64,7 +66,14 @@ public Prel visitLateral(LateralJoinPrel prel, Boolean value) throws RuntimeExce
     children.add(((Prel)prel.getInput(0)).accept(this, value));
     children.add(((Prel) prel.getInput(1)).accept(this, true));
 
-    return (Prel) prel.copy(prel.getTraitSet(), children);
+    if (!value) {
+      return (Prel) prel.copy(prel.getTraitSet(), children);
+    } else {
+      CorrelationId corrId = new CorrelationId(prel.getCorrelationId().getId() + 1);
+      ImmutableBitSet requiredColumns = prel.getRequiredColumns().shift(1);
```

Comment:
Could you please add a comment that shift is required for taking into account newly added `$drill_implicit_field$`